sql/oracle/CMakeLists.txt

```diff
@@ -36,6 +36,6 @@ if(${GLIBCXX_USE_CXX11_ABI})
       COMPILE_DEFINITIONS _GLIBCXX_USE_CXX11_ABI=0)
 endif()
 
-target_compile_definitions(Oracle PUBLIC ${ORACLE_DEFINITIONS})
-target_include_directories(Oracle PUBLIC ${ORACLE_INCLUDE_DIR})
-target_link_libraries(Oracle PUBLIC ${ORACLE_LIBRARIES})
+target_compile_definitions(Oracle PRIVATE ${ORACLE_DEFINITIONS})
+target_include_directories(Oracle PRIVATE ${ORACLE_INCLUDE_DIR})
```

Comment:
```suggestion
target_include_directories(Oracle SYSTEM ${ORACLE_INCLUDE_DIR})
```

Reply:
Why `SYSTEM`

Reply:
These includes should not be exposed to any other tools

Reply:
That'd append `-isystem` which suppresses warnings coming from these headers.

Reply:
Ha, we can do `SYSTEM PRIVATE`

Reply:
Ok, `SYSTEM PRIVATE` make sense then